subctl/diagnore-pre-install.md

```diff
@@ -0,0 +1,113 @@
+# Subctl `diagnose pre-install`
+
+## Summary
+
+As a user of Submariner, it would be useful in a myriad of situations to have a tool capable of testing the Kubernetes infrastructure
+where Submariner is to be installed.
+The tool should check all of Submariner's pre-requisites from the infrastructure, and report any deficiencies.
+It would be nice if the tool can also offer guidance for the deficiencies, so that the user can address them and install Submariner successfully.
+
+This tool would compliment other `subctl diagnose` commands which check that Submariner is installed and working properly, by allowing
+similar checks **before** the actual installation.
+
+Alternatively, the new command could be hosted elsewhere, if nesting under `diagnose` is deemed confusing.
+
+## Proposal
+
+Add a `subctl diagnose pre-install` command which checks that the infrastructure supports installing Submariner by checking:
+
+* Gateway nodes are set-up.
+  * Kube proxy, if there, doesn't run in IPVS mode - as it's currently not supported.
+* The broker is reachable from the cluster.
+* K8s version is supported.
+* The CNI is supported.
+* Conflicts with existing clusters in the clusterset.
+  * CIDRs should be distinct, unless the clusterset is using globalnet.
+* The necessary ports are open.
+  * On the gateway, when not using Load Balancer, the cable driver port(s).
+  * On the nodes, when not using OVN, the vxlan port.
+
+The broker/clusterset checks could be optional and run only if a broker file is supplied.
+
+Alternatively, instead of adding a new command under the `diagnose` commands, these checks could be done as part of`subctl join`.
+This would make sense, since at the point of running the command, we need some information from the user:
```

Comment:
I think this makes sense. Perhaps it's possible a user may want to just check prerequisites w/o actually joining (yet) in  which case we could have an option, eg "check-prerequisites-only".

Reply:
+1 that this makes sense as a part of join